nomad/structs/structs.go

```diff
@@ -11274,11 +11360,12 @@ func (a *Allocation) Expired(now time.Time) bool {
 		return false
 	}
 
-	if tg.MaxClientDisconnect == nil && !tg.PreventRescheduleOnLost {
+	timeout := tg.GetDisconnectLostTimeout()
+	if timeout == 0 && !tg.PreventRescheduleOnLost {
```

Comment:
Do we need to check `tg.Disconnect.Replace` as well? Maybe a helper method would help like for the other fields.

Reply:
We do! my bad